pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -251,6 +261,11 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(env *model.Environme
 					}
 					clusters = append(clusters, subsetCluster)
 				}
+			} else {
+				// Create and apply TLS settings based on destServiceMTLSMode.
+				defaultSni := model.BuildDNSSrvSubsetKey(model.TrafficDirectionOutbound, "", service.Hostname, port.Port)
+				tls := conditionallyConvertToIstioMtls(nil, serviceAccounts, defaultSni, proxy, destServiceMTLSMode)
+				applyUpstreamTLSSettings(env, defaultCluster, tls, proxy.Metadata)
```

Comment:
instead of this logic, its easier to create a placeholder dest rule if the server tls mode is strict. This placeholder rule just needs to have the trafficpolicy.tls.mode set to istio_mutual . Thats it. then you can re-use the rest of the logic and not have to worry about missing out some setting.